tests/integration/e2e/conftest.py

```diff
@@ -241,14 +249,16 @@ async def _deploy_test_app(
         )
 
         # Relate with Kafka
-        await ops_test.model.add_relation(generated_app_name, kafka)
-        await ops_test.model.wait_for_idle(
-            apps=[generated_app_name, kafka], idle_period=30, status="active", timeout=1800
-        )
+        if not integrator:
+            await ops_test.model.add_relation(generated_app_name, kafka)
+            await ops_test.model.wait_for_idle(
+                apps=[generated_app_name, kafka], idle_period=30, status="active", timeout=1800
+            )
```

Comment:
**question:** Why is `integrator` boolean here? What are we trying to do at test runtime? 

Why might we not want an integrator?

Reply:
The variable Integrator represents if we want do the test with the usage of the integrator or not (like we do for tls)

Reply:
In my general comment I'm proposing to move this bit to the tests directly. This may also allow to skip the integrator argument in this function.